src/components/SocialIcon/index.tsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react'
+import Link from '../Link'
+import { ReactComponent as TwitterIcon } from '../../../static/img/community/icon-twitter.svg'
+import { ReactComponent as GithubIcon } from '../../../static/img/community/icon-github.svg'
+import { ReactComponent as LinkedInIcon } from './linkedin.svg'
+
+const icons: { [site: string]: JSX.Element } = {
+  linkedin: <LinkedInIcon />,
+  github: <GithubIcon />,
+  twitter: <TwitterIcon viewBox="5 5 30 30" />
+}
+
+export interface ISocialIcon {
+  url: string
+  site?: string
+}
+
+export interface ISocialIconProps extends ISocialIcon {
+  className?: string
+}
+
+/*
+   Returns a link containing an icon corresponding to the provided site
+
+   Given the situation where either the given link has no site or we don't have
+   an icon for it, we return null such that nothing is rendered in map
+   functions.
+*/
+const SocialIcon: React.FC<ISocialIconProps> = ({
+  site,
+  url,
+  className
+}): JSX.Element | null => {
+  if (!site) return null
```

Comment:
do we need this check? next line will return null or undefined anyway right?

Reply:
Technically no, but this solved a TS issue about using undefined as an index. I'll solve this better with build-time TS.

Reply:
I had to use a combination of `eslint-disable-next-line` and, weirdly enough, some more TS (the `!` at the end of the index) to solve this issue.

From my research, it seems that TS is hard-coded to only accept numbers or strings as object indexes despite what we're doing working fine in vanilla JS. The solution I use now pacifies the TS lint check and doesn't introduce another build-time check.